File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Tanimlamalar/PatientsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { Gender } from "./Gender";
import { PatientsRow } from "./PatientsRow";

export interface PatientsColumns {
    PatientId: Column<PatientsRow>;
    Username: Column<PatientsRow>;
    Name: Column<PatientsRow>;
    Surname: Column<PatientsRow>;
    Gender: Column<PatientsRow>;
    BirthDate: Column<PatientsRow>;
}

export class PatientsColumns extends ColumnsBase<PatientsRow> {
    static readonly columnsKey = 'Tanimlamalar.Patients';
    static readonly Fields = fieldsProxy<PatientsColumns>();
}

[Gender]; // referenced types

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Tanimlamalar/Patients/PatientsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace MuayeneYonetimPortali.Tanimlamalar.Columns;

[ColumnsScript("Tanimlamalar.Patients")]
[BasedOnRow(typeof(PatientsRow), CheckNames = true)]
public class PatientsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignCenter]
    public int PatientId { get; set; }
    public string Username { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Surname { get; set; }
    public Gender Gender { get; set; }
    public DateTime BirthDate { get; set; }
    public string Phone { get; set; }
    public string Email { get; set; }
    public string Tckn { get; set; }
    public IsActive IsActive {get; set; }
}